### ISSUE TITLE
Generates wrong schema for readonly arrays

### ISSUE THREAD

**Issue by mmc41** [YousefED/typescript-json-schema#85]
Very, very nice project! The generated code is however wrong for typescripts buildin ReadonlyArray type. If I generate a schema for BuggyType below it will be very strange. It looks like it does not understand that ReadonlyArray is just like an array except that typescript wont allow it to change.

Nb. tested with typescript 2.1.5 and typescript-json-schema 0.7.0

**Test input:**
```
export interface BuggyType  {
  readonly single: string;
  readonly multiple: ReadonlyArray<string>;
}
```

Wrong output:
```
{
  "type": "object",
  "properties": {
    "single": {
      "type": "string"
    },
    "multiple": {
      "$ref": "#/definitions/ReadonlyArray<string>"
    }
  },
  "additionalProperties": false,
  "required": [
    "single",
    "multiple"
  ],
  "definitions": {
    "ReadonlyArray<string>": {
      "type": "object",
      "properties": {
        "find": {
          "description": "Returns the value of the first element in the array where predicate is true, and undefined\notherwise.",
          "type": "object",
          "properties": {},
          "additionalProperties": false
        },
        "findIndex": {
          "description": "Returns the index of the first element in the array where predicate is true, and -1\notherwise.",
          "type": "object",
          "properties": {},
          "additionalProperties": false
        },
        "length": {
          "description": "Gets the length of the array. This is a number one higher than the highest element defined in an array.",
          "type": "number"
        },
        "toString": {
          "description": "Returns a string representation of an array.",
          "type": "object",
          "properties": {},
          "additionalProperties": false
        },
        "toLocaleString": {
          "type": "object",
          "properties": {},
          "additionalProperties": false
        },
        "concat": {
          "description": "Combines two or more arrays.\nCombines two or more arrays.\nCombines two or more arrays.",
          "type": "object",
          "properties": {},
          "additionalProperties": false
        },
        "join": {
          "description": "Adds all the elements of an array separated by the specified separator string.",
          "type": "object",
          "properties": {},
          "additionalProperties": false
        },
        "slice": {
          "description": "Returns a section of an array.",
          "type": "object",
          "properties": {},
          "additionalProperties": false
        },
        "indexOf": {
          "description": "Returns the index of the first occurrence of a value in an array.",
          "type": "object",
          "properties": {},
          "additionalProperties": false
        },
        "lastIndexOf": {
          "description": "Returns the index of the last occurrence of a specified value in an array.",
          "type": "object",
          "properties": {},
          "additionalProperties": false
        },
        "every": {
          "description": "Determines whether all the members of an array satisfy the specified test.",
          "type": "object",
          "properties": {},
          "additionalProperties": false
        },
        "some": {
          "description": "Determines whether the specified callback function returns true for any element of an array.",
          "type": "object",
          "properties": {},
          "additionalProperties": false
        },
        "forEach": {
          "description": "Performs the specified action for each element in an array.",
          "type": "object",
          "properties": {},
          "additionalProperties": false
        },
        "map": {
          "description": "Calls a defined callback function on each element of an array, and returns an array that contains the results.",
          "type": "object",
          "properties": {},
          "additionalProperties": false
        },
        "filter": {
          "description": "Returns the elements of an array that meet the condition specified in a callback function.\nReturns the elements of an array that meet the condition specified in a callback function.",
          "type": "object",
          "properties": {},
          "additionalProperties": false
        },
        "reduce": {
          "description": "Calls the specified callback function for all the elements in an array. The return value of the callback function is the accumulated result, and is provided as an argument in the next call to the callback function.\nCalls the specified callback function for all the elements in an array. The return value of the callback function is the accumulated result, and is provided as an argument in the next call to the callback function.",
          "type": "object",
          "properties": {},
          "additionalProperties": false
        },
        "reduceRight": {
          "description": "Calls the specified callback function for all the elements in an array, in descending order. The return value of the callback function is the accumulated result, and is provided as an argument in the next call to the callback function.\nCalls the specified callback function for all the elements in an array, in descending order. The return value of the callback function is the accumulated result, and is provided as an argument in the next call to the callback function.",
          "type": "object",
          "properties": {},
          "additionalProperties": false
        }
      },
      "additionalProperties": false,
      "required": [
        "find",
        "findIndex",
        "length",
        "toString",
        "toLocaleString",
        "concat",
        "join",
        "slice",
        "indexOf",
        "lastIndexOf",
        "every",
        "some",
        "forEach",
        "map",
        "filter",
        "reduce",
        "reduceRight"
      ]
    }
  },
  "$schema": "http://json-schema.org/draft-04/schema#"
}
```

**Comment by domoritz** [comment]
Thanks for the bug report! Do you want to create a test and look into a fix? It might be as easy as looking for array type checks (e.g. https://github.com/YousefED/typescript-json-schema/blob/master/typescript-json-schema.ts#L227) and adding `ReadonlyArray`s.

**Comment by mmc41** [comment]
I can try tomorrow if someone is willing to review and merge it quickly if the change is acceptable?

**Comment by domoritz** [comment]
Yes, I'll do that.

**Comment by mmc41** [comment]
Done (I think). Added pull request https://github.com/YousefED/typescript-json-schema/pull/86/commits/8dea21c1f5e1b58a785311d07e947904469123b9

**Comment by domoritz** [comment]
Fixed in 0.8

**Comment by mmc41** [comment]
Thanks for merging and for adding new npm version !
